### ISSUE TITLE
Console displays only response bodies less than 10kb

### ISSUE THREAD

**Issue by Nmirontsev** [a85/POSTMan-Chrome-Extension#7539]
Steps to reproduce:
1. Make a request which will have response body larger than 10kb. 
2. Observe that console doesn't display it and shows the message "The console only shows bodies smaller than 10 KB in size. To view the complete body, inspect it in the Builder window."

In previous versions the limit was 1mb. It became very inconvenient, especially if request is done from pre-request script and there is no way to see response in Builder window.

**Comment by milititskiy** [comment]
+1 Its very inconvenient ! I have several redirects that I can check only in console. Guys, if you have made on aoption " show timestamps" and "hide network" , why you cant add another option "show bodies bigger than 10KB". 
For now, I am  downgrading to the 7.9.0 version. What a pitfull  update...

**Comment by Nmirontsev** [comment]
@milititskiy , how to do a downgrade?

**Comment by milititskiy** [comment]
> @milititskiy , how to do a downgrade?

Uninstalled 7.10 version and downloaded 7.9 There is no other option to resolve this issue. And by the way it is very furstrating, as it seems noone cares about that.

**Comment by Nmirontsev** [comment]
@milititskiy , thank you. New console interface is not good, even not considering 10 kb issue. It should be addressed at some point.

**Comment by milititskiy** [comment]
> @milititskiy , thank you. New console interface is not good, even not considering 10 kb issue. It should be addressed at some point.

You are 100% right! I hope they will hear us. At least they can change 10kb issue.

**Comment by fredrikseiness** [comment]
I too was disapointed to discover this change. I believe the reason given was that large response bodies in the console could cause the tool to crash sometimes. Frustrating as that may be I find the new limit on size even more frustrating. I hope this is altered again soon so that we are able to view larger bodies in the console again.

**Comment by AllanOricil** [comment]
> > @milititskiy , how to do a downgrade?
> 
> Uninstalled 7.10 version and downloaded 7.9 There is no other option to resolve this issue. And by the way it is very furstrating, as it seems noone cares about that.

I could not install the version 7.9 on windows after uninstalling the 7.10
When i try to install 7.9 there is an error message saying "Version Mismatch"

**Comment by AllanOricil** [comment]
> > > @milititskiy , how to do a downgrade?
> > 
> > 
> > Uninstalled 7.10 version and downloaded 7.9 There is no other option to resolve this issue. And by the way it is very furstrating, as it seems noone cares about that.
> 
> I could not install the version 7.9 on windows after uninstalling the 7.10
> When i try to install 7.9 there is an error message saying "Version Mismatch"

There is an issue saying that we can't install a previous version once we have installed a current one.
https://github.com/postmanlabs/postman-app-support/issues/7553

**Comment by fredrikseiness** [comment]
@AllanOricil there is a - slightly painful - way around this. follow these steps to delete all your local files and you will be abel to install version 7.9.0: https://support.getpostman.com/hc/en-us/articles/360025359014-How-to-resolve-the-error-Could-not-open-Postman-

Beware that postman will still download version 7.10.0 and install it when you restart the tool (even if you disable automatic updates)... I have had to do this again twice just to be able to see large bodies in the console (for me it was the less painful option of the two).

**Comment by AllanOricil** [comment]
I found a solution to install a previous version. Create a new windows/Mac  user and install the older version. It will not complain about version mismatch.
But thank you @fredrikseiness for you post.

**Comment by Nmirontsev** [comment]
> I found a solution to install a previous version. Create a new windows/Mac user and install the older version. It will not complain about version mismatch.
> But thank you @fredrikseiness for you post.
@AllanOricil , anyway Postman doesn't allow to control minor version updates. For that reason 7.10 will be downloaded and installed automatically when you relaunch Postman.

**Comment by TravisEaster** [comment]
Yea and here I was about to be happy that the console output scrolls down automatically but the request I need to see is hidden because it's too big... Would really like at least the option to try and see larger requests.

**Comment by milititskiy** [comment]
No updates and news on this issue? We are staying on 7.9.0?

**Comment by saswatds** [comment]
Hi everyone, one of the major features that we added in the new console was to persist logs even when the console was closed, show whitespace characters, etc. This meant that anything we store and show would consume a lot of memory(RAM) and disk-space consumption. The `10kb` limit was a very calculated limit that we put on all **automatically** stored data so that things do not crash and work for majority of our users.

For our power-users we have not put any limit on the amount of data that you can log using `console.*` in the sandbox.
So if you are aware of risks of loading large amount of data you can leverage our sandbox API's in the `Tests` and `Pre-request Script`
```js
console.log(pm.response.text());

// and

console.log(pm.response.json());

// or 

pm.sendRequest("https://postman-echo.com/get", function (err, response) {
    console.log(response.json());
});
```
to log response of any size and also lets lets you even filter-out unnecessary information.

We are aware that this is now an additional step that you need to do, but this would also ensure that
1. The majority of our user community can use the tool without the concern of the tool crashing
2. Our power users can still leverage the features of sandbox and log the bodies of what-ever size they want (Acknowledging the risk involved).

Rest assured, we are continuously working towards improving the performance of the app and in-turn build systems that can handle larger and larger amount of data, but of now this limit will hold.

**Comment by milititskiy** [comment]
> Hi everyone, one of the major features that we added in the new console was to persist logs even when the console was closed, show whitespace characters, etc. This meant that anything we store and show would consume a lot of memory(RAM) and disk-space consumption. The `10kb` limit was a very calculated limit that we put on all **automatically** stored data so that things do not crash and work for majority of our users.
> 
> For our power-users we have not put any limit on the amount of data that you can log using `console.*` in the sandbox.
> So if you are aware of risks of loading large amount of data you can leverage our sandbox API's in the `Tests` and `Pre-request Script`
> 
> ```js
> console.log(pm.response.text());
> 
> // and
> 
> console.log(pm.response.json());
> 
> // or 
> 
> pm.sendRequest("https://postman-echo.com/get", function (err, response) {
>     console.log(response.json());
> });
> ```
> 
> to log response of any size and also lets lets you even filter-out unnecessary information.
> 
> We are aware that this is now an additional step that you need to do, but this would also ensure that
> 
> 1. The majority of our user community can use the tool without the concern of the tool crashing
> 2. Our power users can still leverage the features of sandbox and log the bodies of what-ever size they want (Acknowledging the risk involved).
> 
> Rest assured, we are continuously working towards improving the performance of the app and in-turn build systems that can handle larger and larger amount of data, but of now this limit will hold.

Thank you for reply.
So, let me get it right: you have calculated that 10kb is enough for stable work of your program,after an update that you have made. And before the update there wehere not many complaints on the problem of large responses?
The next thing you are proposing me to use "console.log()" command instead of out of the box funtionality that was present in previous versions? I will not say, that I will have to update all the tests and insert that "crutch" in order to restore previous functionality.
Dont you think that a "win-win" situation will be just adding an option to make (and be aware of crashes) responses bigger that 10kb? Because in previous version , suddenly everthing worked as a charm. 
Most of us realise that "Electron" consumes  alot of RAM and  there is nothing ot do about it, and out of a sudden company starts to think about RAM consumption and starts with one of the most importaint thins in testing - Responses.. C'mon! 
Will be pleased to hear the reasons that you made such questionable descisions on updates. Or to hear true reasons and maybe even help to solve it. And right now it feels like you are not user oriented product any more.

**Comment by Nmirontsev** [comment]
@saswatds , thank you for your response. I understand the reasons behind the decision, but my test suites have thousands of requests. Going through all of them, taking a look which ones have more 10KB response and add console.log to them will be a pain.
It would be more convenient approach if you would add this as a configuration in settings similar to your "Max response size in MB" config. Default value would be 10KB and if anybody wants more they will have an opportunity to change. I started to like extended console functionality, but 10 KB limit is HUGE problem and most users will not like to solve it with console.log.

**Comment by shamasis** [comment]
Just chipping in ... this WILL NOT be limited to 10kb 😊😊 ... in a couple of iterations we want to add an on-demand loading of truncated strings and make this even greater than previous 1mb-ish limit and that too in a controlled non-crashing fashion. What use is console if the aspect you use to debug is itself not showing full info. 😛

**Comment by ntbrierly** [comment]
Just wanted to add my voice - this 10 KB limit is very inconvenient, and I hope something can be done to either increase it or provide some other way to easily view large responses.  It is also a problem when you have several tests that need to be run together because some rely on the results of previous tests, or on variables set in previous tests.  You can't just run one of them in isolation, which is what the error message here seems to be suggesting. Please reconsider this.

**Comment by saswatds** [comment]
Hi everyone, we hear you 😊. Currently, the team is working hard on a solution that will allow loading a considerably large amount of data in bodies in a performant manner using console. We assure you, this will be resolved in a week or two.

Your frustration is absolutely valid but I would request everyone to be a bit patient while we iterate upon this new console interface without compromising on performance.

**Comment by arsieorgano** [comment]
I hope it can be solved as soon as possible... it is not easy to debug if we have such issues. Pro version user here.

**Comment by milititskiy** [comment]
> Hi everyone, we hear you 😊. Currently, the team is working hard on a solution that will allow loading a considerably large amount of data in bodies in a performant manner using console. We assure you, this will be resolved in a week or two.
> 
> Your frustration is absolutely valid but I would request everyone to be a bit patient while we iterate upon this new console interface without compromising on performance.

I am happy that our frustration is making you happy. its been 18 days and you had several updates already, but even canary 13 version stil has restriction of 10kb. Any progress ? Or you just ignore us as therea re not so much compalint s about it?

**Comment by saswatds** [comment]
Hi @milititskiy the app release happens around every 15 days, and solving the technical challenges around providing a solution for loading large amount of data took us some time and we missed a release. The next app version will have the support for showing larger body payloads and more.

**Comment by milititskiy** [comment]
> Hi @milititskiy the app release happens around every 15 days, and solving the technical challenges around providing a solution for loading large amount of data took us some time and we missed a release. The next app version will have the support for showing larger body payloads and more.

Thank you for quick reply. I hope to see things change back to normal.

**Comment by Nmirontsev** [comment]
@saswatds , issue is fixed with 7.13 version. It looks great. Thank you!
